### PR TITLE
Add 404 page to docs site

### PR DESCRIPTION
**Pull request by ross-pfahler**
To test, visit the `error.html` page in the docs build link.

Once merged, any 404s will map to this page.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

react spectrum

### CODE REVIEW

File: packages/dev/docs/pages/error.mdx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+---
+title: Page not found
+---
+
+<Flex alignItems="center" direction="row" height="50vh" justifyContent="center">
+  <IllustratedMessage>
+    <NotFound />
+    <Heading>Error 404: page not found</Heading>
+    <Content>Try reloading this page. If the error persists, contact the React Spectrum team to report it.</Content>
```
Comment (ross-pfahler) [inline packages/dev/docs/pages/error.mdx:27]: I'm not a huge fan of this text; if someone has suggestions im all ears

Comment (dannify) [reply packages/dev/docs/pages/error.mdx:27]: is it for errors? or only 404?
We couldn't find your page? Something went wrong?
Do we even bother saying reload anymore?

Comment (snowystinger) [reply packages/dev/docs/pages/error.mdx:27]: Can we put a link for where to contact us?

Comment (ross-pfahler) [reply packages/dev/docs/pages/error.mdx:27]: good point -- reloads wont matter. its just a 404.

Comment (ross-pfahler) [reply packages/dev/docs/pages/error.mdx:27]: Maybe we say something about choosing a link in the left nav?

File: packages/dev/docs/pages/error.mdx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import NotFound from '@spectrum-icons/illustrations/NotFound';
+
+---
+title: Page not found
+---
+
+<Flex alignItems="center" direction="row" height="50vh" justifyContent="center">
+  <IllustratedMessage>
+    <NotFound />
+    <Heading>Error 404: page not found</Heading>
```
Comment (devongovett) [inline packages/dev/docs/pages/error.mdx:26]:
```suggestion
    <Heading>Error 404: Page not found</Heading>
```

https://spectrum-contributions.corp.adobe.com/page/illustrated-message-beta/#Error:-404

File: packages/dev/docs/pages/error.mdx (hunk 1 above)
Comment (devongovett) [reply packages/dev/docs/pages/error.mdx:27]: You could use the text Spectrum uses in the contribution? https://spectrum-contributions.corp.adobe.com/page/illustrated-message-beta/#Error:-404

```suggestion
    <Content>This page isn't available. Try checking the URL or visit a different page.</Content>
```

File: packages/dev/docs/pages/error.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License. -->
+
+import {Layout} from '@react-spectrum/docs';
+export default Layout;
+
+import {Content} from '@react-spectrum/view';
+ import {Flex} from '@react-spectrum/layout';
```
Comment (ktabors) [inline packages/dev/docs/pages/error.mdx:14]: space

**Comment by ktabors** [comment]
When I enter an invalid url, I don't get this error page. Shouldn't there be configuration for that?

**Comment by ross-pfahler** [comment]
> When I enter an invalid url, I don't get this error page. Shouldn't there be configuration for that?

Yes, see the description
```
Once merged, any 404s will map to this page.
```

**Pull request merged**